src/frontend/apps/impress/src/features/language/api/useChangeUserLanguage.tsx

```diff
@@ -0,0 +1,50 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+
+import { APIError, errorCauses, fetchAPI } from '@/api';
+import { User, useAuthStore } from '@/core';
+
+interface SetUserLanguageParams {
+  language: User['language'];
+}
+
+export const setUserLanguage = async ({
+  language,
+}: SetUserLanguageParams): Promise<User> => {
+  const { userData } = useAuthStore.getState();
+
+  if (!userData?.id) {
+    console.warn('Id of user is needed for this request.');
+    return {} as Promise<User>;
+  }
+
+  const response = await fetchAPI(`users/${userData.id}/`, {
+    method: 'PATCH',
+    body: JSON.stringify({
+      language,
+    }),
+  });
+
+  if (!response.ok) {
+    throw new APIError(
+      `Failed to change the user language to ${language}`,
+      await errorCauses(response, {
+        value: language,
+        type: 'language',
+      }),
+    );
+  }
+
+  return response.json() as Promise<User>;
+};
+
+export function useSetUserLanguage() {
```

Comment:
Let's keep it like the filename.

```suggestion
export function useChangeUserLanguage() {
```